helix-core/src/main/java/org/apache/helix/controller/stages/CurrentStateOutput.java

```diff
@@ -260,13 +260,72 @@ public Message getPendingState(String resourceName, Partition partition, String
     return partitionSet;
   }
 
+  /**
+   * Get the partitions count for each participant with the pending state and given resource state model
+   * @param resourceStateModel specified resource state model to look up
+   * @param state specified pending resource state to look up
+   * @return set of participants to partitions mapping
+   */
+  public Map<String, Integer> getPartitionCountWithPendingState(String resourceStateModel, String state) {
+    Map<String, Integer> pendingPartitionCount = new HashMap<String, Integer>();
+    for (String resource : _pendingStateMap.keySet()) {
+      String stateModel = _resourceStateModelMap.get(resource);
+      if (stateModel != null && stateModel.equals(resourceStateModel)
```

Comment:
I don't think most of us need to google to know that...
But I agree that it could be safer to add parentheses, just to make it safer.
Thanks.